### ISSUE TITLE
The result of rolling aggregations in the presence of NaNs depends changes if bottleneck is installed

### ISSUE THREAD

**Issue by shoyer** [pydata/xarray#776]
Without bottleneck, we are inconsistent with pandas:
```
In [27]: d = xr.DataArray([0, np.nan, 1, 2, np.nan, 3, 4, 5, np.nan, 6, 7], dims='x')

In [28]: d.rolling(x=2).mean()
Out[28]:
<xarray.DataArray (x: 11)>
array([ nan,  0. ,  1. ,  1.5,  2. ,  3. ,  3.5,  4.5,  5. ,  6. ,  6.5])
Coordinates:
  * x        (x) int64 0 1 2 3 4 5 6 7 8 9 10

# using the pandas RC for v0.18
In [29]: d.to_series().rolling(2).mean().to_xarray()
Out[29]:
<xarray.DataArray (x: 11)>
array([ nan,  nan,  nan,  1.5,  nan,  nan,  3.5,  4.5,  nan,  nan,  6.5])
Coordinates:
  * x        (x) int64 0 1 2 3 4 5 6 7 8 9 10
```
xref https://github.com/pydata/xarray/pull/772#discussion_r53556663

**Comment by jhamman** [comment]
I think I know how to fix this. The problem is the define valid windows by their width, not the number of non-null values (`count`). So, we need to add a count and mask step is `Rolling.reduce`.

**Comment by jhamman** [comment]
@shoyer - I think the basic idea for the fix is in #777.  I think after some more rigorous testing (working on it now), it will need a few tweaks, but the idea is there.
